src/providers/twitch/TwitchChannel.cpp

```diff
@@ -808,18 +808,19 @@ void TwitchChannel::loadRecentMessages()
                           messages = std::move(allBuiltMessages)]() mutable {
                 shared->addMessagesAtStart(messages);
 
-                // Notify user about a possible gap in logs
+                // Notify user about a possible gap in logs if it returned some messages
+                // but isn't currently joined to a channel
                 if (QString errorCode = root.value("error_code").toString();
-                    !errorCode.isEmpty())
+                    !errorCode.isEmpty() && messages.size() > 0)
                 {
                     qCDebug(chatterinoTwitch)
                         << QString("rm error_code=%1, channel=%2")
                                .arg(errorCode, this->getName());
                     if (errorCode == "channel_not_joined")
```

Comment:
Let's move the `messages.empty` check here (and make it use isEmpty or empty instead of size() > 0)